src/exceptions/invalidargumentexception.h

```diff
@@ -8,9 +8,9 @@
  */
 class InvalidArgumentException : public QException {
 public:
-    void raise() const { throw *this; }
+    inline void raise() const { throw *this; }
```

Comment:
Are you sure these should be marked as `inline`? Why not just use the regular way `throw InvalidArgumentException` - what essentially is done by using `inline`?

Seems like a very funky way to throw exceptions.

Reply:
"Specifying inline encourages the compiler to do a better job." from http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline.